File: exercises/select_sort/simple_select_sort.py
def select_sort(arr):
    """
    The function provide the simple and not optimistic algorithm
    for sortering array (Selection sort).

    Big O:

        O(n^2)
    """
    for current_index in range(len(arr) - 1):
        min_index = current_index

        for index in range(current_index + 1, len(arr)):
            if arr[min_index] > arr[index]:
                min_index = index
        
        arr[current_index], arr[min_index] = arr[min_index], arr[current_index]

    return arr


if __name__ == '__main__':
    print("\nSelection sort algorithm:\n")

    arr = [1, 4, 3, 5, 7, 2, 4, 1, 2, 3, 5]
    
    print(arr)
    print(select_sort(arr), '\n')

File: exercises/select_sort/simple_select_sort.c
#include <stdio.h>

void select_sort(int arr[], long length)
{
    int min_val;
    int min_index;
    int replace_value;

    for (int i = 0; i <= length - 1; i++) {
        min_val = arr[i];
        min_index = i;

        for (int j = i + 1; j < length; j++) {
            if (arr[j] < min_val) {
                min_val = arr[j];
                min_index = j;
            }
        }

        replace_value = arr[i];

        arr[i] = min_val;
        arr[min_index] = replace_value;
    }
}

void printArrey(int arr[], long length) {
    printf("\narr<");
    for (int i = 0; i < length - 1; i++) {
        printf("%d, ", arr[i]);
    }
    printf("%d>\n", arr[length - 1]);
}

int main()
{
    int arr[] = {1, 3, 5, 10, 11, 2, 4, -1};

    printf("\nSelection sort algorithm:\n");

    printArrey(arr, sizeof(arr) / sizeof(arr[0]));

    select_sort(arr, sizeof(arr) / sizeof(arr[0]));

    printArrey(arr, sizeof(arr) / sizeof(arr[0]));
}
